Update content for S3 persistent storage (#14060)
Lightning-AI/lightning: docs/source-app/glossary/storage/drive_content.rst

> About Drives

Reviewer: ```suggestion
What are Lightning Drives?
```

************

> Lightning Drive storage makes it easy to share files between LightningWorks (Work) so you can run your Lightning App both locally and in the cloud without changing the code.

Reviewer: Add cross references to the relevant classes

Author: I'll do this once I have some examples.

> Lightning Drive storage makes it easy to share files between LightningWorks (Work) so you can run your Lightning App both locally and in the cloud without changing the code.

Reviewer: ```suggestion
Lightning Drives are shared app storage that allow you to share files between LightningWork components, so that you distributed components can share files when running on the cloud. Using drives, you can run your Lightning App both locally and in the cloud without changing the code.
```

The Drive object provides a central place for your components to share data.

> The Drive acts as an isolated folder and any component can access it by knowing its name.

Reviewer: ```suggestion
Lightning Drive can be accessed by any of the app's components by referencing its name.
```

> We currently support two types of Drives: ``lit://`` and ``s3://``.

Reviewer: We should name the types, and then in the table explain how to create each.
```suggestion
We currently support two types of Lighting Drives: Lightning-managed and S3.
```

Reviewer 2: I like this, let’s do “Lightning-managed (``lit://``) and S3 (``s3://``)?

+---------------+-------------------------------------------------------------------------------------------------------------------------------+
| ``lit://``    | ``lit://`` allow read-write operations and are accessible through the Drive API from a Work.                                  |
|               |                                                                                                                               |
|               | They allow your components to put, list, get, and delete files from and to the Drive (except LightningFlows).                 |
+---------------+-------------------------------------------------------------------------------------------------------------------------------+

> | ``s3://``     | ``s3://`` is AWS S3 storage mounted at a filesystem mount point. ``s3://`` is read-only (for now) and its primary purpose is  |
> |               | to give you a permanent location to store your training data.                                                                 |

Reviewer 3: ```suggestion
| ``s3://``     | ``s3://`` is AWS S3 storage mounted at a filesystem mount point. ``s3://`` is read-only (for now) and its primary purpose is  |
|               | to give you a permanent location to access your training data.                                                                 |
```
Since earlier it said read-only, maybe avoid the word "store"?

The Drive acts as an isolate folder and any component can access it by knowing its name.

> Drives offer two types of storage: temporary and persistent.

Reviewer 2: Let's align on these definitions. What we call Temporary here are actually persistent across app restarts, so this nomenclature might generate confusion.

The main difference here is that standard drives are accessible through the Drive API from a Work, while you can mount S3 locations RO to a Work at a filesystem mount point.

This also means that we want to enable mounting RW ("Temporary") drives as RO mount points ("Persistent"), which will make these definitions not work.

Author: @lantiga please take peek. I've made changes based on your feedback.

Think of every instance of the Drive object acting like a Google Drive or like Dropbox.

By sharing the Drive between components through the LightningFlow,
several components can have a shared place to read (``s3://`` Drives) or read and write (``lit://`` Drives) files from.

> Limitations

Reviewer: Create a sub-section only for S3 drives

Author: I'll do this once I get an example

^^^^^^^^^^^

These limitations only apply to ``s3://`` Drives:

> * There is no top level “shareable” S3 drive object. Each S3 Drive is owned by a particular Work.

Reviewer 2: ```suggestion
* There is no top level “shareable” S3 drive object. Each S3 Drive is owned by a particular Work. However, it’s possible to create a Drive with the same location across multiple Works.
```

> * S3 buckets cannot be mounted as Drives at the runtime of a Work. The `Drive` object must be initialized in a Flow and passed to a Work through its initialization arguments.

Reviewer 2: ```suggestion
* S3 buckets cannot be mounted as Drives once a Work has been instantiated. The `Drive` object must be initialized passed to a Work at creation time.
```

> * Whenever a Drive is mounted to a Work, the indexing process will be done again for the provided S3 bucket. This may lead to performance issues with particularly large S3 buckets. For context, 1M files with 2-3 levels of nesting takes less than 1 second to index.

Reviewer 2: “the indexing process” -> “an indexing process” if we didn’t talk about it earlier

Author: Updated

----

**************
Create a Drive
**************

> In order to create a Drive, you simply need to pass its name with the prefix ``lit://`` or ``s3://``.

Reviewer: This doesn't make sense. To create a drive you need to init the class

Reviewer 2: “you simply need to instantiate the Drive object and pass the name of the drive with a lit:// prefix or the location with the s3:// prefix as an argument”.

----

*************************
Transfer files with Drive
*************************

> In the example below, the Drive is created by the Flow and passed to its Work's.

Reviewer 2: Work’s -> Works ?

Author: Updated
